File: longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.go
func lengthOfLongestSubstring(s string) int {
    left, right := 0,0
    longest := 0
    m := make([]bool, 256)
    
    for right < len(s){
        for left < right &&  m[s[right]]{
            m[s[left]] = false
            left += 1
        }
        
        longest = max(right-left +1, longest)
        m[s[right]] = true
        right += 1
    }
    return longest
}

func max(a, b int) int{
    if a >= b{
        return a
    } else{
        return b
    }
}

File: longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        l = res = 0
        charset = set()
        
        for r in range(len(s)):
            while s[r] in charset:
                charset.remove(s[l])
                l += 1
            charset.add(s[r])
            res = max(res, r-l+1)
        return res
            